pkg/lib/operation/repository/checkout/operation.go

```diff
@@ -0,0 +1,32 @@
+package checkout
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/keboola/keboola-as-code/internal/pkg/git"
+	"github.com/keboola/keboola-as-code/internal/pkg/log"
+	"github.com/keboola/keboola-as-code/internal/pkg/model"
+)
+
+const Timeout = 30 * time.Second
+
+type dependencies interface {
+	Logger() log.Logger
+}
+
+func Run(ctx context.Context, repoDef model.TemplateRepository, d dependencies) (*git.Repository, error) {
+	// Create context with timeout
```

Comment:
Checkout operation, v dalsom kroku tu pribudne telemetria (DD span), aby sme to vedeli trackovat v APMku.